core/src/test/java/io/questdb/test/cairo/vm/MemoryCARWSpillableTest.java

```diff
@@ -0,0 +1,1247 @@
+/*******************************************************************************
+ *     ___                  _   ____  ____
+ *    / _ \ _   _  ___  ___| |_|  _ \| __ )
+ *   | | | | | | |/ _ \/ __| __| | | |  _ \
+ *   | |_| | |_| |  __/\__ \ |_| |_| | |_) |
+ *    \__\_\\__,_|\___||___/\__|____/|____/
+ *
+ *  Copyright (c) 2014-2019 Appsicle
+ *  Copyright (c) 2019-2023 QuestDB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.test.cairo.vm;
+
+import io.questdb.cairo.CairoException;
+import io.questdb.cairo.ImplicitCastException;
+import io.questdb.cairo.TableUtils;
+import io.questdb.std.str.Path;
+import io.questdb.test.AbstractCairoTest;
+import io.questdb.test.cairo.TestRecord;
+import io.questdb.cairo.vm.MemoryCARWSpillable;
+import io.questdb.cairo.vm.Vm;
+import io.questdb.cairo.vm.api.MemoryARW;
+import io.questdb.test.griffin.engine.TestBinarySequence;
+import io.questdb.std.*;
+import io.questdb.std.str.StringSink;
+import io.questdb.test.tools.TestUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class MemoryCARWSpillableTest extends AbstractCairoTest {
```

Comment:
There should be some tests that assert the disk spill case (the file gets created when the memory grows and the memory contents are retained, the file gets deleted when the memory shrinks and so on). Also, some negative case tests would be great, e.g. if file creation fails, a meaningful exception is thrown; if file deletion fails on map shrink, an exception is thrown (yet, `FastMap` may ignore it in this scenario as we need to delete the temp dir on startup).